subsys/bluetooth/host/classic/a2dp_codec_sbc.c

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2024 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <string.h>
+#include <errno.h>
+#include <zephyr/sys/atomic.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/sys/printk.h>
+
+#include <zephyr/bluetooth/classic/a2dp_codec_sbc.h>
+#include <zephyr/bluetooth/classic/a2dp.h>
+
+
+uint8_t bt_a2dp_sbc_get_channel_num(void *config_ie)
+{
+	__ASSERT_NO_MSG(config_ie != NULL);
+	struct bt_a2dp_codec_sbc_params *sbc_codec = (struct bt_a2dp_codec_sbc_params *)config_ie;
+
+	if (sbc_codec->config[0] & A2DP_SBC_CH_MODE_MONO) {
+		return 1U;
+	} else if (sbc_codec->config[0] & A2DP_SBC_CH_MODE_DUAL) {
+		return 2U;
+	} else if (sbc_codec->config[0] & A2DP_SBC_CH_MODE_STREO) {
+		return 2U;
+	} else if (sbc_codec->config[0] & A2DP_SBC_CH_MODE_JOINT) {
+		return 2U;
+	} else {
+		return 0U;
+	}
+}
+
+uint32_t bt_a2dp_sbc_get_sampling_frequency(void *config_ie)
+{
+	__ASSERT_NO_MSG(config_ie != NULL);
+	struct bt_a2dp_codec_sbc_params *sbc_codec = (struct bt_a2dp_codec_sbc_params *)config_ie;
```

Comment:
Why is the input parameter type `void *`, yet the implementation assumes a specific struct type (`bt_a2dp_codec_sbc_params`)? How is this API intended to be used in a safe way, since the compiler will not warn if you try to pass a pointer of the wrong type? Same issue with the `get_channel_num()` API.

Reply:
The different codec has different config fields definition. The a2dp.c/h don't need to know the config fields details. For example: a2dp.c/h just callback the config result buffer, for SBC, application use a2dp_codec_sbc.c/h to parse the config result. In my previous implementation, I do the follow way:
```
bt_a2dp_sbc_get_sampling_frequency((struct bt_a2dp_codec_sbc_params *)&codec_cfg->codec_config->codec_ie[0]);
```
I will change it back to this way.

Reply:
Updated.